apps/src/lab2/views/components/Instructions.tsx

```diff
@@ -192,7 +199,11 @@ const InstructionsPanel: React.FunctionComponent<InstructionsPanelProps> = ({
       beforeFinish();
     }
     setIsFinished(true);
-  }, [beforeFinish]);
+
+    if (hocFinishUrl) {
+      shareLab2Project(hocFinishUrl);
+    }
```

Comment:
Should we pass a flag as part of level properties that determines whether the share dialog should be shown or not? Might be good to make it server configured to support non HOC progressions as well?

Reply:
Good call.  I had tried that earlier and have now revived it.